runtime/common/src/lib.rs

```diff
@@ -62,7 +65,43 @@ macro_rules! decl_common_types {
 
         type Address = sp_runtime::MultiAddress<AccountId, ()>;
 
-        type Migrations = (pallet_contracts::Migration<Runtime>,);
+        struct FixStorageVersions;
+
+        impl OnRuntimeUpgrade for FixStorageVersions {
+            fn on_runtime_upgrade() -> frame_support::weights::Weight {
+                StorageVersion::new(4).put::<AdvisoryCommittee>();
+                StorageVersion::new(4).put::<AdvisoryCommitteeMembership>();
+                StorageVersion::new(4).put::<Council>();
+                StorageVersion::new(4).put::<CouncilMembership>();
+                StorageVersion::new(4).put::<TechnicalCommittee>();
+                StorageVersion::new(4).put::<TechnicalCommitteeMembership>();
+                StorageVersion::new(4).put::<Bounties>();
+                StorageVersion::new(1).put::<CampaignAssets>();
+                StorageVersion::new(1).put::<MarketAssets>();
+                StorageVersion::new(1).put::<CustomAssets>();
+                StorageVersion::new(15).put::<Contracts>();
+                StorageVersion::new(1).put::<Balances>(); // TODO
+                <Runtime as frame_system::Config>::DbWeight::get().writes(11)
+            }
+
+            #[cfg(feature = "try-runtime")]
+            fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+                Ok(vec![])
+            }
+
+            #[cfg(feature = "try-runtime")]
+            fn post_upgrade(_: Vec<u8>) -> Result<(), DispatchError> {
+                Ok(())
+            }
```

Comment:
> On runtime upgrade, the on_runtime_upgrade function will clear all storage items associated with the specified pallet, logging the number of keys removed. If the try-runtime feature is enabled, the pre_upgrade and post_upgrade functions can be used to verify the storage removal before and after the upgrade.

What is the number of keys removed? Do you want to verify the storage removal here?

Reply:
It's interesting what the output of try-runtime is. Since pallet-contracts has empty storage for main-net.

Reply:
No, we should trust Parity to implement `RemovePallet` correctly.